File: Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#	 def __init__(self, x):
#		 self.val = x
#		 self.next = None

class Solution:
	# @param head, a ListNode
	# @return a ListNode
	def deleteDuplicates(self, head):
		node = head
		while (node != None):
			next = node.next
			if (next != None and next.val == node.val):
				node.next = next.next
				continue
			node = next
		return head

File: Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode node = head;
        while (node != null) {
            ListNode next = node.next;
            if (next != null && next.val == node.val) {
                node.next = next.next;
                continue;
            }
            node = node.next;
        }
        return head;
    }
}